libs/community/langchain_community/document_loaders/parsers/audio.py

```diff
@@ -12,14 +13,273 @@
+            deployment_name (str):
+                The deployment name of the Whisper model.
+            byte_limit (int):
+                Maximum allowed file size in bytes (default: 25 MB).
+            chunk_duration_threshold (float):
+                Minimum chunk duration for processing, required to be at least
+                0.1 seconds.
+            max_retries (int):
+                Maximum number of retries for failed API requests.
+        """
+        self.api_key = api_key or os.environ.get("AZURE_OPENAI_API_KEY")
+        self.azure_endpoint = azure_endpoint or os.environ.get("AZURE_OPENAI_ENDPOINT")
+        self.api_version = api_version or os.environ.get("OPENAI_API_VERSION")
+        self.azure_ad_token = azure_ad_token
+
+        self.language = language
+        self.prompt = prompt
+        self.response_format = response_format
+        self.temperature = temperature
+
+        self.deployment_name = deployment_name
+        self.chunk_duration_threshold = chunk_duration_threshold
+        self.byte_limit = byte_limit
+        self.max_retries = max_retries
+        self.input_format = input_format
+
+    @property
+    def _create_params(self) -> Dict[str, Any]:
+        params = {
+            "language": self.language,
+            "prompt": self.prompt,
+            "response_format": self.response_format,
+            "temperature": self.temperature,
+        }
+        return {k: v for k, v in params.items() if v is not None}
+
+    def lazy_parse(self, blob: Blob) -> Iterator[Document]:
+        """
+        Lazily parse the provided audio blob for transcription.
+
+        Args:
+            blob (Blob):
+                The audio file in Blob format to be transcribed.
+
+        Yields:
+            Document:
+                Parsed transcription from the audio file.
+
+        Raises:
+            ImportError:
+                If the required packages `openai` or `pydub` are not installed.
+            ValueError:
+                If the file type is unsupported or exceeds the size limit.
+        """
+
+        try:
+            import openai
+        except ImportError:
+            raise ImportError(
+                "openai package not found, please install it with "
+                "`pip install openai`"
+            )
+
+        try:
+            from pydub import AudioSegment
+        except ImportError:
+            raise ImportError(
+                "pydub package not found, please install it with " "`pip install pydub`"
+            )
+
+        if not str(blob.path).endswith(self.input_format):
+            raise ValueError(
+                "File must be of one of the following types: "
+                f"{str(self.input_format)}"
+            )
+
+        if os.path.getsize(str(blob.source)) > self.byte_limit:
+            raise ValueError(
+                "Audio file exceeds 25 MB limit,"
+                f" length of file is {os.path.getsize(str(blob.source))} bytes"
+            )
```

Comment:
Same comment here about client side validation, let's allow the service decide if the request doesn't meet supported parameters.

Reply:
```python
        if os.path.getsize(str(blob.source)) > self.byte_limit:
            raise ValueError(
                "Audio file exceeds 25 MB limit,"
                f" length of file is {os.path.getsize(str(blob.source))} bytes"
            )
```
I included this chunk as when a large file is passed to the api it takes quite a while for an error to be thrown. I have an idea that the transcription starts and the error will be thrown when it hits a limit as opposed to checking beforehand. Let me know if still don't think it's worth it or if there are any better ways to implement this.